hazelcast/src/main/java/com/hazelcast/jet/JetService.java

```diff
@@ -208,6 +209,13 @@ default Job newLightJob(@Nonnull DAG dag) {
      */
     Job newLightJob(@Nonnull DAG dag, @Nonnull JobConfig config);
 
+    /**
+     * For the client side uploads the jar and runs the main method to start the job
+     * For the member side runs the main method of the jar to start the job
```

Comment:
Actually.. if a member calls this method with jar that is not on classpath, then the upload should also be performed, isn't it?

Reply:
HazelcastBootstrap is adding the jar to class path

`URL jarUrl = new URL("file:///" + jar);
 URLClassLoader classLoader = AccessController.doPrivileged(
  (PrivilegedAction<URLClassLoader>) () ->
      new URLClassLoader(new URL[]{jarUrl}, HazelcastBootstrap.class.getClassLoader())
 );`